### PR TITLE
Add GitHub issue forms

### PR DESCRIPTION
**Pull request by mdesmet**
Following PR improves the Create issue form in github.

image

image

image

the last button refers to the #python-client Trino slack channel.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+name: Feature request
+about: Suggest an idea for Trino Python client
+title: ''
+labels: enhancement
+assignees: ''
+
+---
+
+### Describe the feature
+A clear and concise description of what you want to happen.
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/feature_request.md:11]: How about asking for a problem statement first, and then the feature description that should explain how it addresses the problem?

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+### Describe alternatives you've considered
+A clear and concise description of any alternative solutions or features you've considered.
+
+### Additional context
+Please include any other relevant context here.
+
+### Who will this benefit?
+What kind of use case will this feature be useful for? Please be specific and provide examples, this will help us prioritize properly.
+
+### Are you interested in contributing this feature?
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/feature_request.md:22]: Maybe to make the template a bit shorter we should just ask users to assign themselves to the issue. All Trino contributors that signed the CLA are added to the `contributors` group and should be able to do that. If not, just leave a comment.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+contact_links:
+  - name: Ask a question or get support
+    url: https://trinodb.slack.com/channels/python-client
+    about: Ask a question or request support for using Trino Python client
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/config.yml:5]: Add EOL at EOF

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+---
+
+### Describe the bug
+A clear and concise description of what the bug is. What script did you run? What happened?
+
+### Steps To Reproduce
+In as much detail as possible, please provide steps to reproduce the issue. Sample code that triggers the issue, relevant server settings, etc is all very helpful here.
+
+### Expected behavior
+A clear and concise description of what you expected to happen.
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/bug_report.md:17]: I think this should go first, instead of `describe the bug`. After the expected behavior, users should describe the actual behavior. If the bug is not an actual issue, that should make it clear.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+labels: bug
+assignees: ''
+
+---
+
+### Describe the bug
+A clear and concise description of what the bug is. What script did you run? What happened?
+
+### Steps To Reproduce
+In as much detail as possible, please provide steps to reproduce the issue. Sample code that triggers the issue, relevant server settings, etc is all very helpful here.
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/bug_report.md:14]: I think this should be combined with the last section, `System information`. This is so important that maybe even adding a example would be nice, like:
> Using `trino-python-client version X connect to Trino version Y using the connection string 'trino://user@host.example:8080' and see the error message

or
> Using `trino-python-client version X execute the following query on a Trino server version Z:
> ...query and sample data

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+**The output of `import trino; print(trino.__version__)`:**
+```
+<output goes here>
+```
+
+**The operating system you're using:**
+
+**The output of `python --version`:**
+
+### Additional context
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/bug_report.md:32]: This is vague, I'd drop this.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+### Describe the feature
+A clear and concise description of what you want to happen.
+
+### Describe alternatives you've considered
+A clear and concise description of any alternative solutions or features you've considered.
+
+### Additional context
+Please include any other relevant context here.
+
+### Who will this benefit?
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/feature_request.md:19]: Obviously the author :-) I think it's too vague, a problem statement and a solution should be sufficient.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+In as much detail as possible, please provide steps to reproduce the issue. Sample code that triggers the issue, relevant server settings, etc is all very helpful here.
+
+### Expected behavior
+A clear and concise description of what you expected to happen.
+
+### Screenshots and log output
+If applicable, add screenshots or log output to help explain your problem.
+
+### System information
+**The output of `import trino; print(trino.__version__)`:**
```
Comment (hashhar) [inline .github/ISSUE_TEMPLATE/bug_report.md:23]: Should include server version as well.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 9]
```diff
@@ -0,0 +1,5 @@
+---
+contact_links:
+  - name: Ask a question or get support
+    url: https://trinodb.slack.com/channels/python-client
```
Comment (hashhar) [inline .github/ISSUE_TEMPLATE/config.yml:4]: trinodb/trino is trying to use GitHub Discussions - much less friction. For problems which require more interactive troubleshooting we generally just ask people to continue on the Slack.

As a user if you asked me to create an account, click link in email, then add some personal details before being able to ask a question I'd probably not ask it and may decide to not use it altogether.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+---
+
+### Describe the feature
+A clear and concise description of what you want to happen.
+
+### Describe alternatives you've considered
+A clear and concise description of any alternative solutions or features you've considered.
+
+### Additional context
```
Comment (hashhar) [inline .github/ISSUE_TEMPLATE/feature_request.md:16]: Very vague and non-actionable.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+Please include any other relevant context here.
+
+### Who will this benefit?
+What kind of use case will this feature be useful for? Please be specific and provide examples, this will help us prioritize properly.
```
Comment (hashhar) [inline .github/ISSUE_TEMPLATE/feature_request.md:20]: The Trino python client is an open source project and there is no JIRA or task management involved - so there are no priorities. Things will get done when they'll get done and depending on how much effort someone puts in for a certain thing to get done.

cc: @martint

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (hashhar) [reply .github/ISSUE_TEMPLATE/feature_request.md:22]: Yes, just ask people to assign themselves if they're already working on a PR - helps other people who see the issue know to not duplicate work as well.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 9 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/config.yml:4]: We can add the GitHub Discussions as a first option as well and distinguish accordingly.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 10 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/feature_request.md:16]: I will remove this section.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 7 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/feature_request.md:19]: I think this may let the author think more about which persona will benefit from the feature and leave less to the imagination of the person producing the PR.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 6 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/bug_report.md:32]: I will drop this section.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 7 above)
Comment (hashhar) [reply .github/ISSUE_TEMPLATE/feature_request.md:19]: But what would you use that information for?

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/bug_report.md:14]: The advantage of separating it is that it would ensure we have a minimum of information, while free text or an example would shift this responsibility to the user.

Comment (hashhar) [reply .github/ISSUE_TEMPLATE/bug_report.md:14]: But with just server version or client version or OS you can't debug anything - you can guess what went wrong.
So having just System Information without steps is useless.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 4 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/bug_report.md:17]: Good point

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 5 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/bug_report.md:14]: Sure I'm suggesting to keep both sections actually.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/feature_request.md:22]: I think why a lot of other open source projects explicitly ask for this is that it increases the influx of new contributors and shows that the community is there to help. This is probably not so useful for a long time Trino contributor. See eg https://github.com/apache/airflow/issues/new?assignees=&labels=kind%3Abug%2Carea%3Acore&template=airflow_bug_report.yml

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 7 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/feature_request.md:19]: I agree it does not necessarily need a separate section but knowing who the feature is for, is part of a good problem statement. It will help people detect gaps in the feature or may trigger more reflection.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+about: Suggest an idea for Trino Python client
+title: ''
+labels: enhancement
+assignees: ''
+
+---
+
+### Describe the feature
+
+<!-- A clear and concise description of what you want to happen.
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/feature_request.md:12]: This is still missing a "why":
```suggestion
<!-- A clear and concise description of what you want to happen and what problem it would solve. 
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+What kind of use case will this feature be useful for?
```
Comment (nineinchnick) [inline .github/ISSUE_TEMPLATE/feature_request.md:13]: I'm afraid when asked for a use case most people would just repeat the solution instead of defining the problem. We should ask for it explicitly.
```suggestion
<!-- A clear and concise description of what you want to happen
and what problem it would solve.
```

File: .github/ISSUE_TEMPLATE/config.yml (hunk 9 above)
Comment (hashhar) [reply .github/ISSUE_TEMPLATE/config.yml:7]: @martint Can you enable discussions for this repo as well?

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 2 above)
Comment (hashhar) [reply .github/ISSUE_TEMPLATE/feature_request.md:23]: Good point. Let's keep this and we can assign (or tell them to assign themselves). Let's iterate. Doesn't need to be perfect on first go.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 11 above)
Comment (aalbu) [reply .github/ISSUE_TEMPLATE/feature_request.md]: While there may be no formal project management in place, there is some prioritization happening.  A bug leading to incorrect results will get more attention than a cosmetic issue.  Same thing with a feature that has a wider audience vs. a narrow use case.  An issue may be "prioritized" by virtue of getting more contributors interested in implementing it, because of its potential impact.

Comment (leniartek) [reply .github/ISSUE_TEMPLATE/feature_request.md]: Every project, including open source has priorities, bugs vs new features at least.
I don't see any issues with this sentence.

Comment (nineinchnick) [reply .github/ISSUE_TEMPLATE/feature_request.md]: This is about managing expectations. I'd avoid suggesting that any reported issues will have any priority. For non-native English speakers, this might be confusing, even to the point they may think that all issues have _some_ priority, where usually all of them are handled with best effort.

tl;dr let's avoid building up expectations we can't meet.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 9 above)
Comment (martint) [reply .github/ISSUE_TEMPLATE/config.yml:7]: Done

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+      label: Operating System
+      description: What Operating System are you using?
+      placeholder: "You can get it via `cat /etc/os-release` for example"
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Trino Python client version
+      description: Which version of Trino Python client are you using?
+      placeholder: "Execute `import trino; print(trino.__version__)`"
```
Comment (hashhar) [inline .github/ISSUE_TEMPLATE/bug_report.yml:48]: Is this better?

```suggestion
      placeholder: "Execute `python3 -c 'import trino; print(trino.__version__)'`"
```

**Review by hashhar — APPROVED** [review]
A very good start. We can iterate as per feedback over time.

**Comment by hashhar** [comment]
nit: commit and PR should probably say `Add form based GitHub issue flow`/`Add GitHub issue forms`. 😄

**Comment by hashhar** [comment]
Is there some way to "demo" these before I press merge? Or should I merge this into a test repo of mine and check?

**Comment by mdesmet** [comment]
You can check on https://github.com/mdesmet/trino-python-client/issues/new/choose

**Comment by hashhar** [comment]
It's a pity they don't allow copy-pasting the placeholder text. 😞

**Comment by hashhar** [comment]
Thanks for adding this @mdesmet . Let's iterate from here.

**Comment by hashhar** [comment]
@ebyhr PTAL if you want. I'll merge this later tomorrow.

**Review by nineinchnick — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 15]
```diff
@@ -0,0 +1,31 @@
+---
+name: Feature request
```
Comment (ebyhr) [inline .github/ISSUE_TEMPLATE/feature_request.yml:2]: Q There's `Ideas` in GitHub discussions. Do we still want to manage feature requests in issues? 
https://github.com/trinodb/trino-python-client/discussions/categories/ideas

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+      label: Steps To Reproduce
+      description: This will help us reproduce your issue
+      placeholder: >
+        In as much detail as possible, please provide steps to reproduce the issue.
+        Sample code that triggers the issue, relevant server settings, etc is all very helpful here.
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Screenshots and log output
```
Comment (ebyhr) [inline .github/ISSUE_TEMPLATE/bug_report.yml:33]: I don't think suggesting screenshots is a good idea. It would be better to suggest only logs so that we can copy-and-paste for debugging. I agree if this is GUI application though. What do you think about it?

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+      label: Trino Python client version
+      description: Which version of Trino Python client are you using?
+      placeholder: "Execute `python3 -c 'import trino; print(trino.__version__)'`"
+    validations:
+      required: true
+  - type: input
+    attributes:
+      label: Trino Server version
+      description: Which Trino server version are you using?
+      placeholder: "Execute `SELECT VERSION();` on your Trino server"
```
Comment (ebyhr) [inline .github/ISSUE_TEMPLATE/bug_report.yml:55]: No changes requested, "Execute ... on your Trino **_server_**" sounds little weird to me.

**Review by ebyhr — APPROVED** [review]
Looks good except for "screenshots" comment.

File: .github/ISSUE_TEMPLATE/config.yml [hunk 18]
```diff
@@ -0,0 +1,8 @@
+---
+contact_links:
+  - name: Ask a question or get support right here
```
Comment (martint) [inline .github/ISSUE_TEMPLATE/config.yml:3]: We don't provide "support" and we shouldn't create the expectation that people will be able to receive support (which usually has connotations around responsiveness, SLAs, etc. and is associated with commercial ventures). We could change it to "or get help".

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 15 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/feature_request.yml:2]: I think it makes sense to apply formal issue management to feature requests.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 17 above)
Comment (mdesmet) [reply .github/ISSUE_TEMPLATE/bug_report.yml:55]: Maybe "Run" is slightly better?

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 15 above)
Comment (hashhar) [reply .github/ISSUE_TEMPLATE/feature_request.yml:2]: The problem I think Yuya is referring to is that not feature requests are going to be accepted. So we should have some way to filter out what we'll actually work on and what we won't.

Since there's no dedicated backlog other than GitHub issues it makes sense to keep it clean from things that might be asked for but we won't implement (for whatever reasons).

Comment (hashhar) [reply .github/ISSUE_TEMPLATE/feature_request.yml:2]: I'm personally fine with either direction since it seems like premature optimization to me.

**Review by hashhar — APPROVED** [review]

**Comment by hashhar** [comment]
Thank you for being patient and addressing all comments and contributing this. 😃
